packages/editor/src/lib/editor/Editor.ts

```diff
@@ -3858,6 +3869,20 @@ export class Editor extends EventEmitter<TLEventMap> {
 		return this.store.get(typeof asset === 'string' ? asset : asset.id) as TLAsset | undefined
 	}
 
+	async resolveAssetUrl(assetId: TLAssetId | null, context: { zoom: number }): Promise<string> {
+		if (!assetId) return ''
+		const asset = this.getAsset(assetId)
+		if (!asset) return ''
+
+		const networkEffectiveType: string | null =
+			'connection' in navigator ? (navigator as any).connection.effectiveType : null
+		const dpr = this.getInstanceState().devicePixelRatio
```

Comment:
both of these values could in theory be derived pretty easily by the developer - especially if we were to provide `editor` to `onResolveAsset`.

What would it look like to leave deriving these up to implementors? Would that be worth doing for a smaller API?

Reply:
yeah, i was thinking that originally but 2 things:
- i originally (and erroneously was using just `window.devicePixelRatio` whereas i should have been using the instance state version which is more correct
- having these be present on the context give the developer ideas on what they should be thinking about when implementing (a DX reminder, if you will)
- plus it feels like better DX, sort of a nice/helpful butler of an API :)

pits of success as you mention below!

Reply:
makes sense to me!